mediacapture-record/MediaRecorder-stop.html

```diff
@@ -0,0 +1,37 @@
+<!doctype html>
+<html>
+<head>
+    <title>MediaRecorder Stop</title>
+    <link rel="help" href="https://w3c.github.io/mediacapture-record/MediaRecorder.html#mediarecorder">
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+</head>
+<body>
+<canvas id="canvas">
+</canvas>
+<script>
+    function createVideoStream() {
+        let canvas = document.getElementById("canvas");
+        return canvas.captureStream();
+    }
+
+    async_test(t => {
+        let video = createVideoStream();
+        let recorder = new MediaRecorder(video);
+        recorder.onstop = t.step_func(errorEvent => {
+            assert_equals(errorEvent.type, 'stop', 'the error type should be stop');
+            assert_true(errorEvent.isTrusted, 'isTrusted should be true when the event is created by C++');
+            assert_equals(recorder.state, "inactive", "MediaRecorder has been stopped when all tracks are ended");
+            t.done();
+        });
+        assert_equals(video.getVideoTracks().length, 1, "video mediastream starts with one track");
+        recorder.start();
+        assert_equals(recorder.state, "recording", "MediaRecorder has been started successfully");
+        video.getVideoTracks()[0].stop();
+        t.step_timeout(() => {
+            assert_unreached("stop event is not fired after 2 seconds");
+        }, 2000);
```

Comment:
Generally, we prefer just letting the test time out rather than adding an explicit timeout like this.